### ISSUE TITLE
Merging of `InternetExplorerOptions` doesn't work

### ISSUE THREAD

**Issue by JohnA2** [SeleniumHQ/selenium#9143]
## 🐛 Bug Report

This is a follow up for https://github.com/SeleniumHQ/selenium/issues/5279 as requested in https://github.com/SeleniumHQ/selenium/issues/5279#issuecomment-770905915

## To Reproduce

Use `InternetExplorerOptions.merge()`

## Expected behavior

Browser specific parameters in descendants of `MutableCapabilities` work after merging them.

## Test script or set of commands reproducing this issue

```
InternetExplorerOptions options1 = new InternetExplorerOptions();
InternetExplorerOptions options2 = new InternetExplorerOptions();
options1.useCreateProcessApiToLaunchIe().addCommandSwitches("-private");
options1.merge(options2);
WebDriver driver = new InternetExplorerDriver(options1); // IE doesn't start in private mode
```

## Environment

OS: Windows (any version)
Browser: Internet Explorer
Browser version: 11
Browser Driver version: IEDriver 3.150.1
Language Bindings version: Java 4.0.0-alpha-7

**Comment by JohnA2** [comment]
@barancev So the above test code now works correctly in beta-1, but the reverse merge combination (if you replace `options1` with `options2` on line 3 of my code) that worked in alpha-7 is now broken.

**Comment by barancev** [comment]
`merge` method does not modify the object it is called in, it creates a new object that contains merged data.

**Comment by JohnA2** [comment]
ok. Still, this code doesn't work:
```
InternetExplorerOptions options1 = new InternetExplorerOptions();
InternetExplorerOptions options2 = new InternetExplorerOptions();
options2.useCreateProcessApiToLaunchIe().addCommandSwitches("-private");
InternetExplorerOptions options3 = options2.merge(options1);
WebDriver driver = new InternetExplorerDriver(options3); // IE doesn't start in private mode
```

**Comment by barancev** [comment]
A better fix provided by commit 0da772d2b03140fb3378bee12a289a671d1e926f
